core/src/main/java/org/elasticsearch/bootstrap/BootstrapCheck.java

```diff
@@ -249,4 +257,67 @@ long getMaxNumberOfThreads() {
 
     }
 
+    static class MaxMapCountCheck implements Check {
+
+        private final long limit = 1 << 18;
```

Comment:
> it does not make sense to check for 2^16 file descriptors but then 2^18 maps! also by default, we only use mmap directory for certain files so that people dont have to touch this.

This is not correct because memory mapped files are not the only thing that count against `vm.max_map_count`: shared memory segments and mallocs count too.